### DISCUSSION TITLE
Organise Installed Packages in Groups?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5546] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1e560120a65be3b3c7b8a42083d54f795e1d0ec4
Last commit: 2 days ago
Core tap JSON: 07 Aug 10:51 UTC
Core cask tap JSON: 07 Aug 10:51 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I'm not certain if this is a thing that can be done or a feature to propose.

_Most_ packages I have installed are solely in service of another package: `ffmpeg` for `yt-dlp`, `starship` for `fish`, `elixir-ls` for `elixir`, `fd ripgrep cmake libvterm` for `emacs-plus`. What would be very nice is a way to group the ancillary packages under the primary, or just to group the primary and the ancillaries together, so I can see what is for what, and uninstall everything together should the need arise.

**Comment by carlocab** [comment]
Not quite grouping, but you can use `brew leaves` to check for formulae without any dependents installed.

Moreover, if you `brew uninstall emacs-plus`, then that should automatically remove `fd`, `ripgrep`, etc, unless you have `HOMEBREW_NO_AUTOREMOVE` set. If it doesn't do so automatically, you should be able to use `brew autoremove`.

**Reply by ghost** [reply]
I don't mean dependencies, I mean roots. All of the ancillary packages I listed were explicitly installed, by me, to _enhance_ (not enable) the functionality of the primaries. Unless there's a way to locally declare them as dependencies of the primaries, the mentioned tools do not help me.

**Reply by carlocab** [reply]
I see, apologies for misunderstanding. I don't think there's currently a way `brew` handles that. You might be interested in `brew bundle`, though -- it doesn't do exactly what you want, but maybe there's a way to use it that does what you need.

**Reply by ghost** [reply]
Hmm. That's annoying. I'll see if I can finagle `bundle`. Thanks for trying.

**Reply by konnano** [reply]
You might be able to get around this by manually adding the auxiliary package to the depends_on of the primary package expression, but this could create a circular reference.

**Reply by carlocab** [reply]
Seems more sensible to create an [external `brew` command](https://docs.brew.sh/External-Commands) to handle this. You could record the groups you want in JSON, and make that command install and uninstall them together.
